src/number_theory/modular_forms/jacobi_theta.lean

```diff
@@ -92,3 +109,131 @@ begin
+    exact pow_le_pow_of_le_one (exp_pos _).le
+      (exp_le_one_iff.mpr $ (mul_neg_of_neg_of_pos (neg_lt_zero.mpr pi_pos) τ.im_pos).le)
+      (nat.le_self_pow two_ne_zero _) },
+  have s : has_sum (λ n : ℕ, rexp (-π * τ.im) ^ (n + 1))
+    (rexp (-π * τ.im) / (1 - rexp (-π * τ.im))),
+  { simp_rw [pow_succ, div_eq_mul_inv, has_sum_mul_left_iff (real.exp_ne_zero _)],
+    exact has_sum_geometric_of_lt_1 (exp_pos (-π * τ.im)).le
+      (exp_lt_one_iff.mpr $ (mul_neg_of_neg_of_pos (neg_lt_zero.mpr pi_pos) τ.im_pos)) },
+  have aux : summable (λ (n : ℕ), ‖cexp (↑π * I * (↑n + 1) ^ 2 * ↑τ)‖),
+    from summable_of_nonneg_of_le (λ n, norm_nonneg _) this s.summable,
+  exact (norm_tsum_le_tsum_norm aux).trans
+    ((tsum_mono aux s.summable this).trans (le_of_eq s.tsum_eq)),
+end
+
+/-- The norm of `jacobi_theta τ - 1` decays exponentially as `im τ → ∞`. -/
+lemma jacobi_theta_sub_one_is_O_im_infty :
+  is_O upper_half_plane.at_im_infty (λ τ, jacobi_theta τ - 1) (λ τ, rexp (-π * τ.im)) :=
+begin
+  simp_rw [is_O, is_O_with, filter.eventually, upper_half_plane.at_im_infty_mem],
+  refine ⟨2 / (1 - rexp (-π)), 1, (λ τ hτ, (jacobi_theta_sub_one_norm_le τ).trans _)⟩,
+  rw [real.norm_eq_abs, real.abs_exp],
+  refine mul_le_mul_of_nonneg_right _ (exp_pos _).le,
+  rw [div_le_div_left (zero_lt_two' ℝ), sub_le_sub_iff_left, exp_le_exp, neg_mul, neg_le_neg_iff],
+  { exact le_mul_of_one_le_right pi_pos.le hτ },
+  { rw [sub_pos, exp_lt_one_iff, neg_mul, neg_lt_zero], exact mul_pos pi_pos τ.im_pos },
+  { rw [sub_pos, exp_lt_one_iff, neg_lt_zero], exact pi_pos }
+end
+
+-- Formulation of this result is somewhat roundabout, since functions on subtypes don't play well
+-- with `differentiable_at`.
+lemma jacobi_theta_differentiable_at (τ : ℍ) :
+  differentiable_at ℂ (λ z, ∑' (n : ℤ), cexp (π * I * n ^ 2 * z)) ↑τ :=
+begin
+  suffices : ∀ (y : ℝ) (hy : 0 < y),
+    differentiable_on ℂ (λ z, ∑' (n : ℤ), cexp (π * I * n ^ 2 * z)) {z : ℂ | y < im z},
+  from let ⟨y, hy, hy'⟩ := exists_between τ.im_pos in (this y hy).differentiable_at
+    ((complex.continuous_im.is_open_preimage _ is_open_Ioi).mem_nhds (τ.coe_im ▸ hy')),
+  intros y hy z hz,
+  -- Check the easy hypotheses for summability result
+  have h1 : is_open {w : ℂ | y < im w}, from continuous_im.is_open_preimage _ is_open_Ioi,
+  have h2 : is_preconnected {w : ℂ | y < im w}, from (convex_halfspace_im_gt _).is_preconnected,
+  have h3 : ∀ (n : ℤ) (w : ℂ), w ∈ {v : ℂ | y < v.im} →
+    has_deriv_at (λ (v : ℂ), cexp (↑π * I * ↑n ^ 2 * v))
+    (cexp (↑π * I * ↑n ^ 2 * w) * (↑π * I * ↑n ^ 2)) w,
+  { intros n w hw,
+    apply has_deriv_at.cexp,
+    convert (has_deriv_at_id w).const_mul _ using 1,
+    rw mul_one },
+  have h3' := (λ n w hw, (h3 n w hw).has_fderiv_at),
+  have h4 : z ∈ {w : ℂ | y < im w}, from hz,
+  have h5 : summable (λ (n : ℤ), cexp (↑π * I * ↑n ^ 2 * z)),
+    by apply jacobi_theta_summable ⟨z, hy.trans hz⟩,
+  -- The harder one: uniform summability of termwise derivatives
```

Comment:
Normally, you shouldn't need this: a locally uniform limit of holomorphic functions is holomorphic, and we have that in mathlib (`tendsto_locally_uniformly_on.differentiable_on`)

Reply:
OK, I've added a couple of simple lemmas about differentiability of sums of holo functions, and shortened the proof here using those.